src/http/server.rs

```diff
@@ -996,6 +1001,20 @@ pub mod ws {
             }
         }
 
+        pub fn recv_frame_type(&mut self) -> Result<(FrameType, usize), EspError> {
```

Comment:
Did you mean to add this in this PR?

Reply:
I did. I need a way to get the WebSocket frame type and size **without** receiving any data yet, to ensure size invariants and to allocate a buffer big enough for the incoming data.